### PR TITLE
Refactored MockHttpClient to http_spec

### LINKED ISSUE #641: Refactor: Move `MockHttpClient` to `http_spec`
This is a simple refactor to move the implementation of the the MockHttpClient to `http_spec`.

### PR DESCRIPTION
**Pull request by Min2who**
/claim #641
fixed #641

### CODE REVIEW

**Comment by Min2who** [comment]
I have mistakenly commited the gen-root.ts and gen.ts file 
Check my remaining code i will remove those two file later

File: tests/http_spec.rs [hunk 1]
```diff
@@ -150,8 +150,8 @@ impl HttpSpec {
 
 #[derive(Clone)]
 struct MockHttpClient {
-  upstream_mocks: Vec<(UpstreamRequest, UpstreamResponse)>,
-  expected_upstream_requests: Vec<UpstreamRequest>,
+  pub upstream_mocks: Vec<(UpstreamRequest, UpstreamResponse)>,
+  pub expected_upstream_requests: Vec<UpstreamRequest>,
```
Comment (tusharmath) [inline tests/http_spec.rs:154]: do we need this change?

**Comment by tusharmath** [comment]
@Min2who this PR doesn't seem to be doing what the original issue is requesting.

**Comment by tusharmath** [comment]
I am closing this PR because it's not really what the original issue is requests. @Min2who Please feel free to create a new PR with the relevant changes whenever ready. 
Thanks!

**Comment by Min2who** [comment]
sorry for the inconvenience i have created for you @tusharmath 
Have you checked my PR
I have commited mistakenly the npm/gen-root.ts and npm/gen.ts apart from that i guess the PR has solve the problem statement of the issue

**Pull request closed without merging**